client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverAuth.java

```diff
@@ -256,6 +256,104 @@ public void testSuccessFullSslVerification()
         }
     }
 
+    @Test
+    public void testSuccessFullSslVerificationAlternateHostname()
+            throws Exception
+    {
+        String accessToken = newJwtBuilder()
+                .setSubject("test")
+                .setHeaderParam(KEY_ID, "33")
+                .signWith(privateKey33)
+                .compact();
+
+        String url = format("jdbc:trino://127.0.0.1:%s", server.getHttpsAddress().getPort());
+        Properties properties = new Properties();
+        properties.setProperty("SSL", "true");
+        properties.setProperty("SSLVerification", "FULL");
+        properties.setProperty("SSLTrustStorePath", new File(getResource("localhost.truststore").toURI()).getPath());
+        properties.setProperty("SSLTrustStorePassword", "changeit");
+        properties.setProperty("accessToken", accessToken);
+        properties.setProperty("hostnameInCertificate", "localhost");
+
+        try (Connection connection = DriverManager.getConnection(url, properties)) {
+            try (Statement statement = connection.createStatement()) {
+                assertTrue(statement.execute("SELECT 123"));
+                ResultSet rs = statement.getResultSet();
+                assertTrue(rs.next());
+                assertEquals(rs.getLong(1), 123);
+                assertFalse(rs.next());
+            }
+        }
+    }
+
+    @Test
+    public void testFailedFullSslVerificationAlternateHostnameNotProvided()
+            throws Exception
+    {
+        String accessToken = newJwtBuilder()
+                .setSubject("test")
+                .setHeaderParam(KEY_ID, "33")
+                .signWith(privateKey33)
+                .compact();
+
+        String url = format("jdbc:trino://127.0.0.1:%s", server.getHttpsAddress().getPort());
+        Properties properties = new Properties();
+        properties.setProperty("SSL", "true");
+        properties.setProperty("SSLVerification", "FULL");
+        properties.setProperty("SSLTrustStorePath", new File(getResource("localhost.truststore").toURI()).getPath());
+        properties.setProperty("SSLTrustStorePassword", "changeit");
+        properties.setProperty("accessToken", accessToken);
+
+        try (Connection connection = DriverManager.getConnection(url, properties)) {
+            try (Statement statement = connection.createStatement()) {
```

Comment:
nit: can you please merge `try` statements? Same above.

Reply:
Done!